scripts/remove_nearly_empty_products.pl

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/perl -w
+
+# This file is part of Product Opener.
+#
+# Product Opener
+# Copyright (C) 2011-2023 Association Open Food Facts
+# Contact: contact@openfoodfacts.org
+# Address: 21 rue des Iles, 94100 Saint-Maur des Fossés, France
+#
+# Product Opener is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use ProductOpener::PerlStandards;
+
+use CGI::Carp qw(fatalsToBrowser);
+
+use ProductOpener::Config qw/:all/;
+use ProductOpener::Products qw/retrieve_product/;
+use ProductOpener::Store qw/:all/;
+use ProductOpener::Data qw/get_products_collection/;
+
+# This script is run daily to remove empty products (without data or pictures)
+# in particular products created by the button to add a product without a barcode
+
+my $cursor = get_products_collection()->query({data_quality_errors_tags => { '$ne' => ''} })->fields({code => 1});
+$cursor->immortal(1);
+my $removed = 0;
+
+while (my $product_ref = $cursor->next) {
+
+	my $code = $product_ref->{code};
+
+	#print STDERR "updating product $code\n";
+
+	$product_ref = retrieve_product($code);
+
+	no warnings 'uninitialized';
+
+	if ((defined $product_ref) and ($code ne '')) {
+	
+		if (1
+			#and (defined $product_ref->{last_image_t})
+			and ($product_ref->{last_image_t} eq '')
+			and ($product_ref->{owner} eq '')
+			and ($product_ref->{creator} ne 'usda-ndb-import')
```

Comment:
If you put fields like creator, completeness etc. in the retrieved fields of the mongodb query, then you can test their values before you call retrieve_product($code), which saves disk reads

Reply:
Understood, but how can I be sure that it won't time out?